ghost/core/core/server/services/segment/DomainEventsAnalytics.js

```diff
@@ -0,0 +1,82 @@
+const {MilestoneCreatedEvent} = require('@tryghost/milestones');
+
+/**
+ * @typedef {import('@tryghost/logging')} logging
+ */
+
+/**
+ * @typedef {import('analytics-node')} analytics
+ */
+
+/**
+ * @typedef {import('../../../shared/sentry')} sentry
```

Comment:
@allouis Not sure if this is correct, as the sentry module doesn't have types yet... Would it be better to leave it as an `object` instead?

Reply:
So generally speaking you wanna define what you _need_ rather than on what you get.

So rather than depending on concrete implementations, we would depend on an interface.

In this case I would do something like:
```
/**
 * @typedef {object} IExceptionHandler
 * @prop ((err: Error) => void) captureException
 */
```

And then the `sentry` object can satisfy that dependency.

Reply:
You can even then do `this.exceptionHandler.captureException` - which decouples you from sentry completely! NABD though as sentry is deffo pretty core to Ghost atm

Reply:
Cool, did that!! Thanks!